Civi/RemoteEvent/CompilerPass.php

```diff
@@ -0,0 +1,26 @@
+<?php
+
+namespace Civi\RemoteEvent;
+
+use Symfony\Component\DependencyInjection\Compiler\CompilerPassInterface;
+use Symfony\Component\DependencyInjection\ContainerBuilder;
+
+use CRM_RemoteEvent_ExtensionUtil as E;
+
+class CompilerPass implements CompilerPassInterface {
```

Comment:
I'd name it `ActionProviderPass`.

In Symfony universe it's common to put compiler passes in the `DependencyInjection\Compiler` namespace, i.e. `Civi\RemoteEvent\DependencyInjection\Compiler`

Reply:
I would prefer to stick to the documentation of the action-provider-extension for the implementation:
https://lab.civicrm.org/extensions/action-provider/-/blob/master/docs/howto_create_an_action_in_an_extension.md#create-a-compiler-pass-class